src/contexts/PopupContextProvider.tsx

```diff
@@ -23,16 +23,29 @@ export const PopupContextProvider = ({
   children,
 }: PopupContextProviderProps) => {
   const [show, setShow] = useState(false);
+  const [fragment, setFragment] = useState<ReactNode | null>(null);
+
+  const showPopup = (fragmentPiece: ReactNode) => {
+    setFragment(fragmentPiece);
+    setShow(true);
```

Comment:
if you wish to only show the popup whenever the fragment useState has been altered, wouldn't it be more advisable to place an useEffect with 'fragment' on the dependecy array?

Reply:
Why would I do that? The only way that state gets altered is when these two functions are called, we are not exposing the setFragment or setShow functions inside the context. Other than that, if I were to add a useEffect there it would only delay the setState by one render cycle and use extra code.

Reply:
That was just me wondering. The idea is that if you wish to ensure the popup will rise only when the fragment has been updated (which is not guaranteed to happen at the time you call setFragment), a useEffect would be the best solution. If you think the overhead is greater than the benefit (or perhaps the assurance is needless), just ignore my comment.